tests/emulator_tests/persistence_tests/test_shamir_persistence.py

```diff
@@ -16,13 +16,12 @@
 
 import pytest
 
+from tests import buttons
+from tests.emulator_tests.device_handler import BackgroundDeviceHandler
+from tests.emulator_tests.emulators import EmulatorWrapper
+from tests.emulator_tests.upgrade_tests import core_only
```

Comment:
any reason for not using relative imports?

Reply:
No reason. TBH Pycharm did that for me and I'm used to that from `core/`. But I do not have a strong preference. What I would like though is to set a guidance on this. It is another thing we could write to the Contribution Guideline (or random.md or wherever).

Reply:
hmm, true, pycharm does do that :)

I would prefer relative imports especially here, given that `tests` itself is not a meaningful module from the outside.

Not sure about core. My preference would be relative imports as well, but I don't know why the current code style exists, so maybe i'll need to find that out first.

Reply:
> I would prefer relative imports especially here, given that tests itself is not a meaningful module from the outside.

The thing is `tests.buttons` and `tests.device_handler` is now in the root and the code is two folders away. I can't import that relatively, can I? There is no such thing as `from .... import buttons` is there?

> Not sure about core. My preference would be relative imports as well, but I don't know why the current code style exists, so maybe i'll need to find that out first.

I don't really care that much, but I like how consistent it is. Especially since some modules are named the same (get_address, sign_tx etc.) you always now, what you are importing. I believe we used it differently everywhere, which was quite messy, so we've unified it.

Reply:
> > I would prefer relative imports especially here, given that tests itself is not a meaningful module from the outside.
> 
> The thing is `tests.buttons` and `tests.device_handler` is now in the root and the code is two folders away. I can't import that relatively, can I? There is no such thing as `from .... import buttons` is there?

Yes there is, you can add as many dots as you like. Also it should be `...` if i'm counting correctly.
But I agree that relative imports this deep are not great either.

Which brings me to think: do we actually need the `upgrade_tests` to live apart from `persistence_tests`? Shouldn't both simply be in `emulator_tests`?

Reply:
But the persistence_test at the moment does not test other firmware versions. It is independent of all that. There will be another persistence test then in upgrade_tests that deals with that.

Reply:
Yes, but my point was more of, how many upgrade tests will there ever be? Can't they be differentiated by being called `test_upgrade_*` or something?

Or do you want separate targets for CI?

Reply:
...because in that case, I'm not sure if common nesting under "emulator tests" is doing what we want either.